plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/discoverer/XcpServerDiscoverer.java

```diff
@@ -410,6 +409,8 @@ else if (prodBrand.equals("XenServer") && prodVersion.equals("6.1.0"))
             return new XenServer610Resource();
         else if (prodBrand.equals("XenServer") && prodVersion.equals("6.5.0"))
             return new XenServer650Resource();
+        else if (prodBrand.equals("XenServer") && prodVersion.equals("7.0.0"))
+            return new XenServer700Resource();
```

Comment:
@syed from my tests, I found that prodversion will always be 6.5 for any xen with version >6.4(which is done in CitrixHelper.getProductVersion()). I would suggest removing this new class XenServer700Resource(as we dont do any thing special here) and recording the correct xen version.
I have done those changes on top of this PR at  https://github.com/Accelerite/cloudstack/commit/648d8edc98ec37c6f6a2f84ea4b0a63a77fc35fa
Please go through them and if you agree, we can make it part of this PR